component/src/main/java/io/siddhi/extension/io/grpc/util/ServiceConfigs.java

```diff
@@ -49,15 +52,19 @@
     private String keystoreAlgorithm;
     private String tlsStoreType;
     private boolean isSslEnabled;
+     Logger log = LoggerFactory.getLogger(ServiceConfigs.class.getName());
 
-    public ServiceConfigs(OptionHolder optionHolder, SiddhiAppContext siddhiAppContext, String streamID) {
+    public ServiceConfigs(OptionHolder optionHolder, SiddhiAppContext siddhiAppContext,
+                          String streamID, ConfigReader configReader) {
         if (optionHolder.isOptionExists(GrpcConstants.RECEIVER_URL)) {
             this.url = optionHolder.validateAndGetOption(GrpcConstants.RECEIVER_URL).getValue();
-        } else if (optionHolder.isOptionExists(GrpcConstants.PUBLISHER_URL)) {
-            this.url = optionHolder.validateAndGetOption(GrpcConstants.PUBLISHER_URL).getValue();
+            log.debug("GRPC Service for : " + streamID + " started");
         } else {
-            throw new SiddhiAppValidationException(siddhiAppContext.getName() + ": " + streamID + ": either " +
-                    "receiver.url or publisher.url should be given. But found neither");
+            this.url = configReader.readConfig(GrpcConstants.RECEIVER_URL , GrpcConstants.DEFAULT_RECEIVER_URL);
```

Comment:
As discussed offline, Receiver URL cannot be system properties, use siddhi references for that